prae/src/core.rs

```diff
@@ -101,32 +101,50 @@ where
         self.0
     }
 
+    /// Verifies invariants. This is guaranteed to succeed unless you've used
+    /// one of the `unsafe` methods that require variants to be manually upheld.
+    pub fn verify(&self) -> Result<(), E> {
+        G::validate(&self.0).map_or(Ok(()), Err)
+    }
+}
+
+#[cfg(feature = "unchecked-access")]
+trait UncheckedAccess {
     /// Construct a value without calling `adjust` and `validate`. The invariant must be upheld
     /// manually. Should be used only for optimisation purposes.
-    pub unsafe fn new_manual<V: Into<T>>(v: V) -> Self {
+    pub fn new_unchecked<V: Into<T>>(v: V) -> Self;
+
+    /// Mutate a value without calling `adjust` and `validate`. The invariant must be upheld
+    /// manually. Should be used only for optimisation purposes.
+    pub fn mutate_unchecked(&mut self, f: impl FnOnce(&mut T));
+
+    /// Gives mutable access to the internals without upholding invariants.
+    /// They must continue to be upheld manually while the reference lives!
+    pub fn get_mut(&mut self) -> &mut T;
+}
+
+#[cfg(feature = "unchecked-access")]
+impl UncheckedAccess for Guard {
+    /// Construct a value without calling `adjust` and `validate`. The invariant must be upheld
+    /// manually. Should be used only for optimisation purposes.
+    unsafe fn new_unchecked<V: Into<T>>(v: V) -> Self {
```

Comment:
I thought we agreed that `unsafe` should not be used?

Reply:
All removed now.